start-mycroft.sh

```diff
@@ -180,80 +184,80 @@ function check-dependencies() {
 _opt=$1
 _force_restart=false
 shift
-if [[ "${1}" == "restart" ]] || [[ "${_opt}" == "restart" ]] ; then
+if [ "${1}" = "restart" ] || [ "${_opt}" = "restart" ] ; then
     _force_restart=true
-    if [[ "${_opt}" == "restart" ]] ; then
+    if [ "${_opt}" = "restart" ] ; then
         # Support "start-mycroft.sh restart all" as well as "start-mycroft.sh all restart"
         _opt=$1
     fi
     shift
 fi
```

Comment:
Sorry I just realised we also run into the same problem here. It's not intended behaviour to use this script without args, but I'd rather we show the help info if someone does try to use it incorrectly.

I guess the easiest method would be to throw a conditional on both? But it's getting late here so I may just be getting lazy...

Reply:
The key bit being "It's not intended behaviour to use this script without args".

I checked if `$#` equals 0 now, and if it does show the help and exit. If it's at least 1 or more, both shifts should be successful now I think?

Reply:
The other edge case would be `mycroft-start restart`
which now fails with a shift error when called using start-mycroft.sh
and directs to update dependencies for bin/mycroft-start

Reply:
I'm not sure how that still happens but ok, I added a check for shift within the restart block too.